nh_odoo_fixes/models/datetime_utils.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""Contains various useful methods for managing datetimes."""
+from datetime import datetime
+
+from openerp import models
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT as DTF
+
+
+class DatetimeUtils(models.AbstractModel):
+
+    _name = 'datetime_utils'
+
+    @classmethod
+    def zero_microseconds(cls, date_time):
+        """
+        Return the passed date_time with any microseconds set to 0.
+
+        :param date_time:
+        :type date_time: datetime or str
+        :return:
+        """
+        if isinstance(date_time, str):
```

Comment:
Could this not be achieved easier with the use of the `datetime.strptime` -> change millseconds to 0 -> `datetime.strftime`